expo/expo
# [expo-camera] Prevent shutter sound when device volume is muted

This update modifies the takePicture function to check the device's media volume before playing the shutter click sound. If the volume is set to zero, the sound will not be played. This change ensures that the app respects the user's sound settings, enhancing user experience by preventing unexpected noise when the device is muted.

# Why

This PR aims to enhance user experience by respecting the audio settings of the user's device. Currently, the application plays a shutter sound even when the device is muted, which can be unexpected and disruptive. This change ensures that the shutter sound is played only if the device's volume is not muted, adhering to user settings more closely. This adjustment is particularly important in environments where minimizing noise is essential.

# How

The implementation involves checking the current volume level of the device using the AudioManager class before triggering the shutter sound. If the volume is set to zero (muted), the sound will not play. This logic is added right before the MediaActionSound().play(MediaActionSound.SHUTTER_CLICK) call within the onCaptureStarted() method of the ImageCapture.OnImageCapturedCallback() interface.

# Test Plan

To verify these changes, follow these steps:

- Set the device's media volume to zero.
- Open the application and use the feature to take a picture.
- Observe that no shutter sound is played.

Repeat the test with the device's media volume at non-zero levels to ensure the shutter sound plays as expected.


# Checklist
